container/kvm/cache.go

```diff
@@ -0,0 +1,362 @@
+package kvm
+
+import (
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	yaml "gopkg.in/yaml.v1"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/environs/imagedownloads"
+	"github.com/juju/juju/environs/simplestreams"
+	"github.com/juju/juju/juju/paths"
+	"github.com/juju/utils/series"
+)
+
+const (
+	// FType is the file type we want to fetch and use for kvm instances.
+	FType = "disk1.img"
+)
+
+// Oner is an interface which allows us to use sync params to call
+// imagedownloads.One or pass in a fake for testing.
+type Oner interface {
+	One() (*imagedownloads.Metadata, error)
+}
+
+// SyncParams conveys the information necessary for calling imagedownloads.One.
+type SyncParams struct {
+	arch, series, ftype string
+	srcFunc             func() simplestreams.DataSource
+}
+
+// One implemnets Oner.
+func (p SyncParams) One() (*imagedownloads.Metadata, error) {
+	return imagedownloads.One(p.arch, p.series, p.ftype, p.srcFunc)
+}
+
+// Validate that our local type fulfull their implementations.
+var _ Oner = (*SyncParams)(nil)
+var _ Updater = (*updater)(nil)
+
+// Updater is an interface to permit faking input in tests. The default
+// implementation is updater, defined in this file.
+type Updater interface {
+	Update() error
+}
+
+// updater is our default Updater implementation.
+type updater struct {
+	md        *imagedownloads.Metadata
```

Comment:
Does it have to be? Familiarity permits brevity a la, jc.ErrIsNil, gc.Matches. Particularly here. It is only used in the internal implementation here. FWIW I embiggened it anyways.